carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_guidance

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import rospy
+from cav_msgs.msg import RouteEvent
+from cav_msgs.msg import Plugin
+
+route_selected = False
+plugins_activated = False
+
+def check_route_status(route_event_msg):
+    if route_event_msg.event == 1:
+        print("Route Selected")
+        route_selected = True
+
+def check_plugin_status(plugin_msg):
+    if plugin_msg.activated == True:
+        print("Plugins Activated")
+        #NOTE: May have to add logic to parse through every individual plugin and determine their status
+        plugins_activated = True
+
+def initialize():
+    rospy.init_node("carma_carla_guidance")
+    service = rospy.ServiceProxy('/set_active_guidance')
+    #While-loop monitors the route and plugin status and sets guidance to active once both are confirmed true
+    while rospy.is_shutdown() != True:
+        route_sub = rospy.Subscriber("/route_event",RouteEvent(), check_route_status)
+        plugin_sub = rospy.Subscriber("/plugin_discovery", Plugin(), check_plugin_status)
+        if route_selected == True:
+            if plugins_activated == True:
+                service.call(True) #Sets guidance status to active
+                break
+    
+    rospy.spin()
```

Comment:
I think the rospy.spin() could be removed since the loop takes care of its work.